### PR TITLE
feat: Expose Globals registry to glimmerx

### PR DESCRIPTION
**Pull request by chadhietala**
While GlimmerX is a module-centric world, applications can still have
macros that are defined by the surrounding build system. This was
partially there with `config.additionalGlobals` but we actually
need to expose a registry that can be merged into so that the
globals aren't `any` types

### CODE REVIEW

**Review by chriskrycho — APPROVED** [review]
Seems reasonable to me but I'll defer to @dfreeman before merging it!

**Review by dfreeman — APPROVED** [review]
Looks good to me.

My internet has gone out, so I won't be able to do a release. @chriskrycho will you have the chance to do that today?

**Comment by chriskrycho** [comment]
Should be able to! Merging it now, will try to release in a bit.

**Pull request merged**
